wagtail/snippets/models.py

```diff
@@ -40,19 +41,38 @@ def get_edit_url(self, instance):
             )
 
 
-def register_snippet(model):
+def register_snippet(model, viewset=None):
     if model not in SNIPPET_MODELS:
+        from wagtail.snippets.views.chooser import SnippetChooserViewSet
+        from wagtail.snippets.views.snippets import SnippetViewSet
+
         model.get_usage = get_object_usage
         model.usage_url = get_snippet_usage_url
         model.get_admin_base_path = get_admin_base_path
         model.get_admin_url_namespace = get_admin_url_namespace
-        SNIPPET_MODELS.append(model)
-        SNIPPET_MODELS.sort(key=lambda x: x._meta.verbose_name)
 
-        url_finder_class = type(
-            "_SnippetAdminURLFinder", (SnippetAdminURLFinder,), {"model": model}
+        if viewset is None:
+            viewset = getattr(model, "admin_viewset", SnippetViewSet)
```

Comment:
We can also remove this since it was never documented...

Reply:
yep, I agree we should remove this. Do you want to make this in a separate PR?

Reply:
I think it makes sense to remove it in this PR, so that this PR effectively replaces how the viewset for a snippet model is found and initialised. I've added a new commit for this. Thanks!